defaultclient.go

```diff
@@ -122,64 +126,98 @@ func NewDefaultClient(config *Config) Client {
 
 	debug = config.Debug
+
 	log("NewDefaultClient: debug enabled")
 
 	return client
 }
 
 // ClientTokenGrant starts client token grant to get client bearer token for role caching
-func (client *DefaultClient) ClientTokenGrant() error {
-	refreshInterval, err := client.clientTokenGrant()
+func (client *DefaultClient) ClientTokenGrant(opts ...Option) error {
+	options := processOptions(opts)
+	span, _ := jaeger.StartSpanFromContext(options.jaegerCtx, "client.ClientTokenGrant")
+
+	defer jaeger.Finish(span)
+
+	refreshInterval, err := client.clientTokenGrant(span)
 	if err != nil {
+		jaeger.TraceError(span, err)
+
 		return logAndReturnErr(
 			errors.WithMessage(err,
 				"ClientTokenGrant: unable to do token grant"))
 	}
 
 	go func() {
 		client.tokenRefreshActive = true
+
 		time.Sleep(refreshInterval)
-		client.refreshAccessToken()
+		client.refreshAccessToken(span)
 	}()
 
 	log("ClientTokenGrant: token grant success")
+
 	return nil
 }
 
 // ClientToken returns client access token
-func (client *DefaultClient) ClientToken() string {
+func (client *DefaultClient) ClientToken(opts ...Option) string {
+	options := processOptions(opts)
+	span, _ := jaeger.StartSpanFromContext(options.jaegerCtx, "client.ClientToken")
+
+	defer jaeger.Finish(span)
+
 	return client.clientAccessToken
 }
 
 // StartLocalValidation starts goroutines to refresh JWK and revocation list periodically
 // this enables local token validation
-func (client *DefaultClient) StartLocalValidation() error {
-	err := client.getJWKS()
+func (client *DefaultClient) StartLocalValidation(opts ...Option) error {
+	options := processOptions(opts)
+	span, _ := jaeger.StartSpanFromContext(options.jaegerCtx, "client.StartLocalValidation")
+
+	defer jaeger.Finish(span)
+
+	err := client.getJWKS(span)
 	if err != nil {
+		jaeger.TraceError(span, errors.WithMessage(err,
+			"StartLocalValidation: unable to get JWKS"))
+
 		return logAndReturnErr(
 			errors.WithMessage(err,
 				"StartLocalValidation: unable to get JWKS"))
 	}
 
-	err = client.getRevocationList()
+	err = client.getRevocationList(span)
 	if err != nil {
+		jaeger.TraceError(span, errors.WithMessage(err,
+			"StartLocalValidation: unable to get revocation list"))
+
 		return logAndReturnErr(
 			errors.WithMessage(err,
 				"StartLocalValidation: unable to get revocation list"))
 	}
 
-	go client.refreshJWKS()
-	go client.refreshRevocationList()
+	go client.refreshJWKS(span)
+
+	go client.refreshRevocationList(span)
 
 	client.localValidationActive = true
 
 	log("StartLocalValidation: local validation activated")
+
 	return nil
 }
 
 // ValidateAccessToken validates access token by calling IAM service
-func (client *DefaultClient) ValidateAccessToken(accessToken string) (bool, error) {
+func (client *DefaultClient) ValidateAccessToken(accessToken string, opts ...Option) (bool, error) {
+	options := processOptions(opts)
+	span, _ := jaeger.StartSpanFromContext(options.jaegerCtx, "client.ValidateAccessToken")
```

Comment:
Is it safe if options.jaegerCtx is nil?

Reply:
should be safe https://github.com/AccelByte/iam-go-sdk/pull/15/files#diff-334c015efe845cd85942e06a5b5c75b2R197